megamek/src/megamek/common/Aero.java

```diff
@@ -897,6 +927,32 @@ public void newRound(int roundNumber) {
 
         // update velocity
         setCurrentVelocity(getNextVelocity());
+        
+        //If we're an aero on an atmospheric map, remove space-only sensors from the list of options
```

Comment:
I appreciate the attention paid to building in the flexibility to handle future changes. But as written it will take the time to review all sensors of all units every round when playing on a space map, whether other maps are in place or not. When movement between space and atmospheric maps becomes a possibility, there will need to be adjustments made to any Entity that switches map type (such as selecting an initial hex along the edge of the new map), and that would be a better place for this. I recommend putting it in its own method which is called at some point prior to deployment on the map, which makes it simple enough to call during map transition.